docs/api/auth.md

```diff
@@ -0,0 +1,134 @@
+# Auth API
+
+This documentation describes the API endpoints for user authentication and authorization in our fest management framework.
+
+## Base URL
+
+`/api/auth`
+
+## Endpoints
+
+### `POST` /login
+
+Authenticate a user with email and password.
+
+#### Request
+
+<details>
+<summary> Toggle Request Body </summary>
+
+```json
+{
+  "email": "user@example.com",
+  "password": "user_password"
+}
+```
+
+</details>
+Response
+<details>
+<summary> Toggle Response </summary>
+
+```json
+{
+  "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI1MzQ5YjRkZGQyNzgxZDA8c2FsdHJvdW5kcyFuZGhhZGhhcHBlcnMiLCJyb2xlIjoidXNlciIsImlhdCI6MTYzNDkyOTk4MSwiZXhwIjoxNjcwOTIyOTgxLCJvcmdhbmlzYXRpb24iOiI1ZjdmMmI5YTRmM2QxZTBiMWM5YjRjN2EiLCJyb2xlIjoiVXNlciBDaGV0dHJpIn0.WYsdP4_U8SVsC9vV9vyG_BrG3A3gD9GkN0m1i6S7auk",
+  "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI1MzQ5YjRkZGQyNzgxZDA4YzA5ODRmMyIsInJvbGUiOiJ1c2VyIiwiaWF0IjoxNjM0OTI5OTgxLCJleHAiOjE2NzA5MjI5ODF9.WYsdP4_U8SVsC9vV9vyG_BrG3A3gD9GkN0m1i6S7auk",
```

Comment:
Refresh token won't be provided via response json

Reply:
so should i remove this field altogether?

Reply:
Yes, You can write info or something similar to responses and requests etc. A new subtitle like "Response Cookies"

Reply:
hey @jindalujjwal0720 , i did some research on the controllers/**auth.js** file

**For login()**
> const { user } = req.body;
      const { email, password } = user;
      const {
        accessToken,
        refreshToken,
        user: userPayload,
      } = await AuthService.loginWithEmailPassword(email, password);
    
**For register()**
>const { user } = req.body;
      const {
        accessToken,
        refreshToken,
        user: userPayload,
      } = await AuthService.register(user);

**For refresh()**
>const { festifyRefreshToken } = req.cookies;
      const {
        accessToken,
        refreshToken,
        user: userPayload,
      } = await AuthService.refreshAccessToken(festifyRefreshToken);

from the code , what i can infer is that in **login()** and **register()**,
the request is from the **req.body**

while in **refresh()**, the request is from **req.cookies**

Also in all these functions, the response is same:
>const {
        accessToken,
        refreshToken,
        user: userPayload,
      } 


so i changed only the request for the **refresh()** part:
earlier it was : 
> ### `GET` /refresh
Refresh an access token using a refresh token.

Request
<details>
<summary> Toggle Request Body</summary>

```json
{
  "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI1MzQ5YjRkZGQyNzgxZDA4YzA5ODRmMyIsInJvbGUiOiJ1c2VyIiwiaWF0IjoxNjM0OTI5OTgxLCJleHAiOjE2NzA5MjI5ODF9.WYsdP4_U8SVsC9vV9vyG_BrG3A3gD9GkN0m1i6S7auk"

}
```
</details>

-------------------------------------------------------------------------------------------------------------------------

Now i changed it to: 
> ## `GET` /refresh
Refresh an access token using a refresh token.

Request
<details>
<summary> Toggle Request Cookies </summary>

```json
{
  "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI1MzQ5YjRkZGQyNzgxZDA4YzA5ODRmMyIsInJvbGUiOiJ1c2VyIiwiaWF0IjoxNjM0OTI5OTgxLCJleHAiOjE2NzA5MjI5ODF9.WYsdP4_U8SVsC9vV9vyG_BrG3A3gD9GkN0m1i6S7auk"

}
```
</details>

Reply:
![image](https://github.com/we-festify/festify/assets/69349350/b5d0085e-2924-4eff-8a78-73451f3772a2)

i hope this is ok?
i read the controllers/auth.js file and it gave me clarification on what to do.
thanks for the valuable info